cobigen-cli/cli-systemtest/src/test/java/com/devonfw/cobigen/cli/systemtest/GenerateCommandIT.java

```diff
@@ -65,6 +68,68 @@ public void generateFromEntityTest() throws Exception {
         .exists();
   }
 
+  /**
+   *
+   * Test the upgrade process from templates in a given path and generates from the new template-sets structure
+   *
+   * @throws Exception test fails
+   */
+  @Test
+  @Ignore // TODO: re-enable when upgrade process is implemented, see: https://github.com/devonfw/cobigen/issues/1595
+  public void upgradeAndGenerateFromEntityTest() throws Exception {
+
+    FileUtils.copyDirectory(new File(testFileRootPath + "templatesproject"), this.tmpProject.toFile());
+    this.tmpProject.resolve("templates-devon4j").toFile()
+        .renameTo(this.tmpProject.resolve(ConfigurationConstants.COBIGEN_TEMPLATES).toFile());
+    File baseProject = this.tmpProject.resolve("maven.project/core/").toFile();
+    File monolithicConfiguration = this.tmpProject.toFile();
+    String args[] = new String[7];
+    args[0] = "generate";
+    args[1] = this.entityInputFile.getAbsolutePath();
+    args[2] = "--increments";
+    args[3] = "0";
+    args[4] = "--upgrade-configuration";
+    args[5] = "-tp";
+    args[6] = monolithicConfiguration.getAbsolutePath();
+
+    execute(args, false);
+
+    assertThat(this.currentHome.resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_PATH)).exists();
+
+    assertThat(baseProject.toPath().resolve("src/main/java/com/maven/project/sampledatamanagement/logic/api/to"))
+        .exists();
+  }
+
+  /**
+   *
+   * Todo
+   *
+   * @throws Exception test fails
+   */
+  @Test
+  @Ignore // TODO: re-enable/fix when CLI tests were re-factored, see: https://github.com/devonfw/cobigen/issues/1659
+  public void generateFromEntityWithTemplateSetTest() throws Exception {
+
+    File baseProject = this.tmpProject.resolve("maven.project/core/").toFile();
+    File templateSetsConfig = this.tmpProject.resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER).toFile();
+    String args[] = new String[6];
+    args[0] = "generate";
+    args[1] = this.entityInputFile.getAbsolutePath();
+    args[2] = "--increments";
+    args[3] = "0";
+    args[4] = "-tp";
+    args[5] = templateSetsConfig.getAbsolutePath();
+
+    execute(args, false);
+
+    assertThat(this.currentHome.resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_PATH)).exists();
+
+    assertThat(baseProject.toPath().resolve("src/main/java/com/maven/project/sampledatamanagement/logic/api/to"))
+        .exists();
+  }
+
+  // TODO: Add similar test with a template-set jar file in downloaded folder, see: https://github.com/devonfw/cobigen/issues/1661
```

Comment:
Create an issue and reference it or remove the TODO

Reply:
done.